src/index.ts

```diff
@@ -1,154 +1,4 @@
-        const variableName = toCamelCase(constructorName)
-        let attempt = 1
-        componentName = variableName
-        while (componentMap.takenNames.has(componentName)) {
-          attempt++
-          componentName = `${variableName}_${attempt}`
-        }
-        const componentCode = this.writeComponent(constructorName, componentInstance)
-        componentMap.takenNames.add(componentName)
-        componentMap.instanceToName.set(componentInstance, componentName)
-        code += `const ${componentName} = ${componentCode}\n`
-      }
-      code += `${name}.addComponentOrReplace(${componentName})\n`
-    }
-
-    if (!parent) {
-      code += `engine.addEntity(${name})\n`
-    }
-
-    return code
-  }
-
-  private getConstructorName(component: DCL.ObservableComponent) {
-    // Get class name
-    let constructor: string
-    const typesArray = Object.keys(this.map.exports).filter(
-      n => this.map.exports[n].kind === 'class' && n !== 'Shape' && n !== 'ObservableComponent'
-    )
-    for (let i = 0; i < typesArray.length; i++) {
-      if (component instanceof this.DCL[typesArray[i]]) {
-        constructor = typesArray[i]
-      }
-    }
-    return constructor
-  }
-
-  private writeComponent(constructorName: string, component: DCL.ObservableComponent) {
-    return constructorName === 'Transform'
-      ? `new Transform(${this.getTransformComponentData(component.data)})`
-      : `new ${constructorName}(${this.getConstructorValues(constructorName, component.data)})`
-  }
-
-  private getEntityName(entity: DCL.IEntity) {
-    let entityName: string
-
-    this.entities.forEach((ent, name) => {
-      if (ent === entity) {
-        entityName = name
-      }
-    })
-
-    return entityName
-  }
-
-  private getConstructorValues(constructor: string, data) {
-    const types = this.getConstructorTypes(constructor)
-
-    if (!types || !types.parameters) {
-      return ''
-    }
-
-    const values = Object.keys(types.parameters).map(p => {
-      if (types.parameters[p].type === 'number') {
-        return data[p]
-      }
-
-      return `'${data[p]}'`
-    })
-    return values.join(', ')
-  }
-
-  private getConstructorTypes(name: string) {
-    if (!this.map.exports[name].members) {
-      return null
-    }
-
-    return this.map.exports[name].members.__constructor
-  }
-
-  private getTransformComponentData(data): string {
-    const props = []
-    for (const prop in data) {
-      if (data[prop] instanceof this.DCL.Vector3) {
-        props.push(`${prop}: new Vector3(${data[prop].x}, ${data[prop].y}, ${data[prop].z})`)
-      }
-
-      if (data[prop] instanceof this.DCL.Quaternion) {
-        props.push(
-          `${prop}: new Quaternion(${data[prop].x}, ${data[prop].y}, ${data[prop].z}, ${
-            data[prop].w
-          })`
-        )
-      }
-    }
-    return `{\n  ${props.join(',\n  ')}\n}`
-  }
-}
+export { SceneWriter }
+export default SceneWriter
```

Comment:
Why are we exporting the writer twice? I'd prefer to only export it once to be consistence, I don't really have a preference if default or not.

Reply:
I think the `default` export was there before this PRs, so removing it would break backwards, but since the builder is the only dependant on this package i think we can remove it and just convert it to a named import on the builder side